lib/config.js

```diff
@@ -229,6 +231,16 @@ function getLocalConfig(thisConfig, directory) {
             continue;
         }
 
+        // Check for root flag
+        if (localConfig.root === true) {
+            rootPath = path.dirname(localConfigFile);
+        }
+
+        // If root flag is set, don't consider file if it is above root
+        if (rootPath && !pathIsInside(path.dirname(localConfigFile), rootPath)) {
```

Comment:
I always get jittery when extra unnecessary work is being done, there's more potential for error.

I could be over thinking it, though. Nevermind. :)